src/interaction/FixedLocalTupleComListInteractionTemplate.hpp

```diff
@@ -0,0 +1,408 @@
+			}
+		    }
+		    pidK++;
+		}
+
+		Real3D* totCOM;
+		totCOM = new Real3D[num_of_subchain];
+		Real3D* COM;
+		COM = new Real3D[num_of_subchain];
+		
+		Real3D Li = system.bc->getBoxL();
+		
+		for(long unsigned int i = 0; i < num_of_subchain; i++) {
+		    COM[i] = 0.;
+		    totCOM[i] = 0.;
+		}
+
+		for (FixedLocalTupleList::TupleList::Iterator it(*fixedtupleList); it.isValid(); ++it) {
+		    Particle *p = it->first;
+		    Real3D pos_i = p->position();
+		    Real3D tmp_com = pos_i;
+		    std::vector<Particle*> pList = it->second;
+
+		    for (long unsigned int j = 1; j < N_Constrain; j++) {
+			p = pList[j - 1];
+			Real3D pos_j = p->position();
+			Real3D dist;
+			system.bc->getMinimumImageVectorBox(dist, pos_j, pos_i);
+			pos_j = pos_i + dist;
+			tmp_com += pos_j;
+			pos_i = pos_j;
+		    }
+		    COM[(p->id() - 1)/N_Constrain] = tmp_com/N_Constrain;
+		}
+
+		boost::mpi::all_reduce(*system.comm, COM, num_of_subchain, totCOM, std::plus<Real3D>() );
+		
+		for (long unsigned int i = 0; i < num_of_subchain; i++) {
+		    com_origin[i] = totCOM[i];
+		}
+		
+		delete [] COM;
+		COM = NULL;
+		delete [] totCOM;
+		totCOM = NULL;
+	    }
+	    
+	    virtual ~FixedLocalTupleComListInteractionTemplate() {};
+	    
+	    // set the center of mass of subchains
+	    void setCom(longint id, const Real3D& pos) {
+		com_origin[id] = pos;
+	    }
+
+	    void
+	    setFixedLocalTupleList(shared_ptr < FixedLocalTupleList > _fixedtupleList) {
+		fixedtupleList = _fixedtupleList;
+	    }
+	    
+	    shared_ptr < FixedLocalTupleList > getFixedLocalTupleList() {
+		return fixedtupleList;
+	    }
+	    
+	    void
+	    setPotential(shared_ptr < Potential> _potential) {
+		if (_potential) {
+		    potential = _potential;
+		} else {
+		    LOG4ESPP_ERROR(theLogger, "NULL potential");
+		}
+	    }
+	    
+	    shared_ptr < Potential > getPotential() {
+		return potential;
+	    }
+	    
+	    virtual void addForces();
+	    virtual real computeEnergy();
+	    virtual real computeEnergyDeriv();
+	    virtual real computeEnergyAA();
+	    virtual real computeEnergyCG();      
+	    virtual void computeVirialX(std::vector<real> &p_xx_total, int bins); 
+	    virtual real computeVirial();
+	    virtual void computeVirialTensor(Tensor& w);
+	    virtual void computeVirialTensor(Tensor& w, real z);
+	    virtual void computeVirialTensor(Tensor *w, int n);
+	    virtual real getMaxCutoff() { return 0.0; }
+	    virtual int bondType() { return Nonbonded; }
+	    
+	protected:
+	    std::map<long unsigned int, Real3D> computeCOM() {
+
+		std::map<long unsigned int, Real3D> com;
+
+		const bc::BC& bc = *getSystemRef().bc;  // boundary conditions
+
+		for (FixedLocalTupleList::TupleList::Iterator it(*fixedtupleList); it.isValid(); ++it) {
+		    Particle *p = it->first;
+		    Real3D pos_i = p->position();
+		    Real3D tmp_com = pos_i;
+		    std::vector<Particle*> pList = it->second;
+
+		    for (long unsigned int j = 1; j < N_Constrain; j++) {
```

Comment:
I modified "j = 1" and "j < N_constrain" to "j = 0" and "j < N_constrain - 1" respectively.
Since, `pList` contain (N_constrain - 1) particles according with the specification of `FixedLocalTupleList`.